machine/fdt.c

```diff
@@ -144,6 +144,23 @@ int fdt_string_list_index(const struct fdt_scan_prop *prop, const char *str)
   return -1;
 }
 
+void fdt_version_prop_print(const uint32_t * version_value, int version_len)
+{
+  const char *char_data = (char *)version_value;
+  int char_len = version_len;
+  if (!char_data) {
+    char_data = "unknown";
+    char_len = strlen(char_data) + 1;
+  }
+  // The size should be 1, but print any extra values if they appear
+  for (int i = 0; i < char_len; i += strlen(char_data + i) + 1) {
```

Comment:
```suggestion
  if (!char_data) {
    char_data = "unknown";
    version_len = strlen(char_data) + 1;
  }
  // The size should be 1, but print any extra values if they appear
  for (int i = 0; i < version_len; i += strlen(char_data + i) + 1) {
```

No need for the char_len variable?